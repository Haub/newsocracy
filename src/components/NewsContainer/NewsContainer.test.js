import React from 'react';
import { shallow } from 'enzyme';
import { NewsContainer } from './index.js';
jest.mock('uuid', () => jest.fn(() => 1));


it('should match the snapshot with props', () => {
  const mockArticles = [{ title: 'Name', source: 'npr.org'}];
  const wrapper = shallow(<NewsContainer articles={mockArticles} />);
  expect(wrapper).toMatchSnapshot();
});