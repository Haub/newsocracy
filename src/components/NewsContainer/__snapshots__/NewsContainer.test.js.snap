// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsContainer
    articles={
      Array [
        Object {
          "source": "npr.org",
          "title": "Name",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="left column"
        >
          <h2
            className="left-title"
          >
            From the Left:
          </h2>
          <NewsTitle
            article={
              Object {
                "source": "npr.org",
                "title": "Name",
              }
            }
          />
        </section>,
        <section
          className="center column"
        >
          <h2
            className="center-title"
          >
            From the Center:
          </h2>
        </section>,
        <section
          className="right column"
        >
          <h2
            className="right-title"
          >
            From the Right:
          </h2>
        </section>,
      ],
      "className": "news-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="left-title"
            >
              From the Left:
            </h2>,
            Array [
              <NewsTitle
                article={
                  Object {
                    "source": "npr.org",
                    "title": "Name",
                  }
                }
              />,
            ],
          ],
          "className": "left column",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "From the Left:",
              "className": "left-title",
            },
            "ref": null,
            "rendered": "From the Left:",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "article": Object {
                "source": "npr.org",
                "title": "Name",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="center-title"
            >
              From the Center:
            </h2>,
            Array [],
          ],
          "className": "center column",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "From the Center:",
              "className": "center-title",
            },
            "ref": null,
            "rendered": "From the Center:",
            "type": "h2",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="right-title"
            >
              From the Right:
            </h2>,
            Array [],
          ],
          "className": "right column",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "From the Right:",
              "className": "right-title",
            },
            "ref": null,
            "rendered": "From the Right:",
            "type": "h2",
          },
        ],
        "type": "section",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="left column"
          >
            <h2
              className="left-title"
            >
              From the Left:
            </h2>
            <NewsTitle
              article={
                Object {
                  "source": "npr.org",
                  "title": "Name",
                }
              }
            />
          </section>,
          <section
            className="center column"
          >
            <h2
              className="center-title"
            >
              From the Center:
            </h2>
          </section>,
          <section
            className="right column"
          >
            <h2
              className="right-title"
            >
              From the Right:
            </h2>
          </section>,
        ],
        "className": "news-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="left-title"
              >
                From the Left:
              </h2>,
              Array [
                <NewsTitle
                  article={
                    Object {
                      "source": "npr.org",
                      "title": "Name",
                    }
                  }
                />,
              ],
            ],
            "className": "left column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "From the Left:",
                "className": "left-title",
              },
              "ref": null,
              "rendered": "From the Left:",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "article": Object {
                  "source": "npr.org",
                  "title": "Name",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="center-title"
              >
                From the Center:
              </h2>,
              Array [],
            ],
            "className": "center column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "From the Center:",
                "className": "center-title",
              },
              "ref": null,
              "rendered": "From the Center:",
              "type": "h2",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="right-title"
              >
                From the Right:
              </h2>,
              Array [],
            ],
            "className": "right column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "From the Right:",
                "className": "right-title",
              },
              "ref": null,
              "rendered": "From the Right:",
              "type": "h2",
            },
          ],
          "type": "section",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
